### DISCUSSION TITLE
Loading Data for streaming next video when phone is locked (especially on Android)

### DISCUSSION THREAD

**Discussion by sgebr01** [TheWidlarzGroup/react-native-video#4230] [Q&A]
### What are you trying to do?

I'm trying to make my app fetch the next video's streaming details and then start playing that video. I'm wondering if anybody has any tips as to where I should put this logic? I want it to continue loading in the background like Spotify and other music streaming apps do, especially when the phone is locked.

### What have you tried?

Currently I'm doing it in an effect with a dependency on the video's id and making the API request there as well, but if the phone is locked and screen off, the request doesn't go through since the phone is in "doze mode" on Android.

### Platforms

Android

**Comment by freeboub** [comment]
unfortunnatelly I don't think it it feature.
I know exoplayer have such method to optimize loading of next content, but this is not integrated ...

**Comment by TWG-AI** [comment]
For the video to continue playing when the phone is locked, your app needs the `audio` background mode enabled in `Info.plist` (`UIBackgroundModes: [audio]`). The currently-playing video must have an audio track (or be explicitly configured for background audio) — iOS won't keep a purely-video stream alive in the background.

For preloading the *next* video's streaming details while the current one plays: react-native-video doesn't have a built-in preload-metadata API, but you can fetch your next video's manifest URL at the app level using `fetch` while the current video is playing. The actual media preloading (video.js-style `preload()`) is available in v7 via `VideoPlayer.preload(source)`.
